internal/pkg/service/stream/plugin/throttle.go

```diff
@@ -0,0 +1,27 @@
+package plugin
+
+import (
+	"context"
+
+	"github.com/keboola/keboola-as-code/internal/pkg/service/stream/definition/key"
+	targetModel "github.com/keboola/keboola-as-code/internal/pkg/service/stream/storage/level/target/model"
+)
+
+type throttleFn func(ctx context.Context, sinkKey key.SinkKey) bool
+
+type Throttle struct {
+	SinkKey  key.SinkKey
+	Provider targetModel.Provider
+}
+
+func (p *Plugins) RegisterThrottle(provider targetModel.Provider, fn throttleFn) {
+	p.throttle[provider] = fn
+}
+
+func (p *Plugins) IsThrottled(ctx context.Context, provider targetModel.Provider, sinkKey key.SinkKey) bool {
```

Comment:
I think you can use it in both, why would you do from something generic more specific thing? You can use it anywhere you would like to evaluate it. I don't uderstand why I would do `CanAcceptNewFileRotation` and than change it back to `IsThrottled` because I used it in `sliceRotation` for example.

Reply:
The point is that if slice upload also cared about this, we'd need to use a separate method anyway so that it can be decided independently. The fact that the bridge is unable to accept another file import does not imply it is unable to accept a slice upload (and vice-versa).

Reply:
577e20d055b0c7249651d3067ff12e376fce763e

Reply:
Much better though I'd drop the Rotation part... `CanAcceptNewFileRotation` => `CanAcceptNewFile`. The plugin doesn't care about rotations, it just decides whether it can process another file.

Reply:
0ef0afedab52a89f2926daa5da154d8502c6477c